packages/ra-data-graphql-simple/src/buildGqlQuery.ts

```diff
@@ -21,17 +21,65 @@ import getFinalType from './getFinalType';
 import isList from './isList';
 import isRequired from './isRequired';
 
+type SparseFields = (string | { [k: string]: SparseFields })[];
+type ExpandedSparseFields = { linkedType?: string; fields: SparseFields }[];
+
+function processSparseFields(
```

Comment:
Overall, I find the `processSparseFields` algorithm a bit redundant: there are a bunch of loops to get fields name for instance. I think it can be simplified.